SimG4Core/Generators/src/Generator.cc

```diff
@@ -163,6 +164,9 @@ void Generator::HepMC2G4(const HepMC::GenEvent *evt_orig, G4Event *g4evt) {
         // be propagated by GEANT, so do not change their status code.
         status = 2;
       }
+      if (status == 2 && abs(pdg) == 9900015) {
```

Comment:
@mgratti , would it be more correct to have following lines:

// heavy neutrino should not be propagated by Geant4
 
if(std::abs(pdg) == 9900015) status = 3

Reply:
here if decay vertex of heavy neutrino will be inside fiductial volume (probability is low but still possible) then decay vertex of it will not be created but secondary particles will be created.

Reply:
@civanch, Sorry for the delay in replying to your comments.

For my case, it does not make a difference because the production vertex of the HNL is anyway created because there is another status=1 particle that is a child of that vertex.

I thought of potential other use case: for example a signal with two HNLs produced at a vertex and no other status=1 particle (extremely rare). If both HNLs decay within the fiducial region, depending on the HNL status, the production vertex will (status=3) or will not (status=2) be saved as a G4 vertex. However, this will have no impact in the end; in fact the HNL will have status=3 in the second part (to avoid G4 propagation) and, as a result, no particle will be associated to this G4 vertex.